src/test/java/io/druid/data/input/MapBasedRowTest.java

```diff
@@ -51,4 +56,24 @@ public void testGetLongMetricFromString()
     Assert.assertEquals(-9223372036854775807L, row.getLongMetric("k5"));
     Assert.assertEquals(9223372036854775802L, row.getLongMetric("k6"));
   }
+
+  @Test
+  public void testMapWithNull()
+  {
+    Map<String, Object> map = new HashMap<>();
+    map.put("k0", null);
+    map.put("k1", Collections.singletonList(null));
+    map.put("k2", Arrays.asList(null, null, null));
+    MapBasedRow row = new MapBasedRow(new DateTime(), map);
+
+    // A) if value is NULL, then an empty list is returned
+    // (actually, its if (!map.containsKey()), with java8
+    // one should use getOrDefault)
+    Assert.assertEquals(Collections.emptyList(), row.getDimension("k0"));
+
+    // B) if value is a list, then _any_ NULL value is transformed to string "null"
```

Comment:
My bad, I think we can leave this for now to document the existing behavior and we'll fix it in a separate PR.